cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -281,15 +281,22 @@ private predicate exprToExprStep_nocfg(Expr fromExpr, Expr toExpr) {
   or
   toExpr = any(StmtExpr stmtExpr | fromExpr = stmtExpr.getResultExpr())
   or
-  toExpr = any(Call call |
-      exists(DataFlowFunction f, FunctionInput inModel, FunctionOutput outModel, int iIn |
-        call.getTarget() = f and
-        f.hasDataFlow(inModel, outModel) and
-        outModel.isOutReturnValue() and
-        inModel.isInParameter(iIn) and
-        fromExpr = call.getArgument(iIn)
+  exists(Call call, DataFlowFunction f, FunctionInput inModel, FunctionOutput outModel |
+    call.getTarget() = f and
+    f.hasDataFlow(inModel, outModel) and
+    (
+      exists(int argInIndex |
+        inModel.isInParameter(argInIndex) and
+        fromExpr = call.getArgument(argInIndex)
+      ) or (
+        inModel.isInQualifier() and
+        fromExpr = call.getQualifier()
       )
+    ) and (
+      outModel.isOutReturnValue() and
+      toExpr = call
     )
+  )
```

Comment:
I'm slowly adapting to our new coding style, but to be honest my brain is still mostly wired with the old one.  I'm noticing that the philosophy is actually quite different - it seems to be all about communicating the logical structure of the code at a glance now, rather than giving clues about the intended meaning.

(I've adopted your suggestion, and fixed some merge issues; still other comments to address though)